packages/blade/src/components/Popover/_decisions/decisions.md

```diff
@@ -0,0 +1,275 @@
+| onOpenChange    | `({ isOpen }) => void`                                                                                                 | `undefined` | Called when popover isOpen state is changed, this can be used to detect when popover opens or closed                                  |          |
+| initialFocusRef | `React.RefObject`                                                                                                      | `undefined` | The ref of the element that should receive focus when the popover opens.                                                              |          |  |
+
+### Examples:
+
+#### Usage
+
+```js
+// basic example
+<Popover
+  placement="top-end"
+  headerTitle="International Payments"
+  content={
+    <Text>
+      Your business can go international with support for transactions in 
+      <Link href="#">100 foreign currencies.</Link>
+    </Text>
+  }
+>
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Popover>
+```
+
+<img src="./assets/popover-example-1.png" alt="Example usage of popover" width="70%" />
+
+#### Controlled 
+
+```js
+const ControlledPopover = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  return (
+    <>
+    <Button onClick={()=> setIsOpen(prev => !prev)}>Trigger</Button>
+    <Popover
+      isOpen={isOpen}
+      onOpenChange={({ isOpen }) => setIsOpen(isOpen)}
+      content={<Text>Any content</Text>}
+    >
+      <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+    </Popover>
+    </> 
+  )
+}
+```
+
+#### Uncontrolled 
+
+```js
+const UncontrolledPopover = () => {
+  return (
+    <Popover
+      defaultIsOpen={true}
+      content={<Text>Any content</Text>}
+    >
+      <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+    </Popover>
+  )
+}
+```
+
+#### On initial load 
+
+To open the popover on initial load, consumers can just pass `defaultIsOpen` prop and whenever the popover mounts it will open.
+
+```js
+const InitialOpen = () => {
+  return (
+    <Popover
+      defaultIsOpen={true}
+      content={<Text>Any content</Text>}
+    >
+      <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+    </Popover>
+  )
+}
+```
+
+#### Working with non-interactive triggers
+
+To make popover apear on clicking a non-interactive elements such as icons, badges, counters etc. We will provide a `PopoverInteractiveWrapper` component which will work as a minimal trigger:
+
+```js
+import { Popover, PopoverInteractiveWrapper } from '@razorpay/blade';
+
+// non-interactive element as trigger
+<Popover content="Amount reversed to customer bank account">
+  <PopoverInteractiveWrapper>
+    <Counter value={100} />
+  </PopoverInteractiveWrapper>
+</Popover>;
+```
+
+> Note: If users don't wrap the non-interactive elements in `PopoverInteractiveWrapper` the popover won't apear
+
+### Custom Triggers
+
+Users will be able to use their own custom components as triggers:
```

Comment:
```suggestion
Users will be able to use their own custom interactive components(mention the names of these components) as triggers:
```

Reply:
> (mention the names of these components)

Anything they like. as long as they expose the refs and events.

Reply:
but then how will we diff between interactive and non-interactive and when to put the wrapper for making the underlying element interactive?

Reply:
PopoverInteractiveWrapper is a convenience wrapper that we provide to consumers, technically it is a "Custom Trigger" just that we expose this custom trigger so that consumer doesn't have to wire up the custom trigger for each time they are using a non-interactive element since the wiring up code is quite boilerplate-y.  

Also imagine this case: 
https://github.com/razorpay/blade/issues/1574

```jsx
    <Tooltip content="Hello world" placement="bottom">
      <TooltipInteractiveWrapper>
        <Button isDisabled={true}>Hover over me</Button>
      </TooltipInteractiveWrapper>
    </Tooltip>
```

If a Button is disabled, it won't respond to any events (onClick, mouseDown etc), in this case if you want to open the popover you need to wrap it in PopoverInteractiveWrapper.